python/tvm/relay/frontend/onnx.py

```diff
@@ -1899,12 +1899,28 @@ def _impl_v1(cls, inputs, attr, params):
         )
         return _op.reshape(onehot, shape_of(inputs[0]))
 
+    @classmethod
+    def _impl_v13(cls, inputs: OnnxInput, attr: typing.Dict, params: typing.Dict) -> relay.Expr:
+        ndim = len(infer_shape(inputs[0]))
+        axis = attr.get("axis", -1) % ndim
+        dtype = infer_type(inputs[0]).checked_type.dtype
```

Comment:
Note that it would be more efficient to only call `infer_type` once then extract `ndim` and `dtype` from that.

Reply:
good call, updated